_zengl.py

```diff
@@ -386,51 +389,50 @@ def validate(attributes, uniforms, uniform_buffers, vertex_buffers, layout, reso
         elif name in uniform_buffer_map:
             uniform_buffer_binding_map[binding] = obj
         else:
-            raise ValueError('Cannot set layout binding for "{}"'.format(name))
+            raise ValueError(f'Cannot set layout binding for "{name}"')
 
     for obj in uniforms:
         name = obj['name']
         location = obj['location']
         if location < 0:
             continue
         if name not in layout_map:
-            raise ValueError('Missing layout binding for "{}"'.format(name))
+            raise ValueError(f'Missing layout binding for "{name}"')
         binding = layout_map[name]['binding']
         if binding not in sampler_resources:
-            raise ValueError('Missing resource for "{}" with binding {}'.format(name, binding))
+            raise ValueError(f'Missing resource for "{name}" with binding {binding}')
 
     for obj in uniform_buffers:
         name = obj['name']
         if name not in layout_map:
-            raise ValueError('Missing layout binding for "{}"'.format(name))
+            raise ValueError(f'Missing layout binding for "{name}"')
         binding = layout_map[name]['binding']
         if binding not in uniform_buffer_resources:
-            raise ValueError('Missing resource for "{}" with binding {}'.format(name, binding))
+            raise ValueError(f'Missing resource for "{name}" with binding {binding}')
 
     for obj in resources:
         resource_type = obj['type']
         binding = obj['binding']
         if resource_type == 'uniform_buffer':
             buffer = obj['buffer']
             if binding not in uniform_buffer_binding_map:
-                raise ValueError('Uniform buffer binding {} does not exist'.format(binding))
+                raise ValueError(f'Uniform buffer binding {binding} does not exist')
             name = uniform_buffer_binding_map[binding]['name']
             if binding in bound_uniform_buffers:
-                raise ValueError('Duplicate uniform buffer binding for "{}" with binding {}'.format(name, binding))
+                raise ValueError(f'Duplicate uniform buffer binding for "{name}" with binding {binding}')
             size = uniform_buffer_map[name]['size']
             if buffer.size < size:
-                msg = 'Uniform buffer is too small {} is less than {} for "{}" with binding {}'
-                raise ValueError(msg.format(buffer.size, size, name, binding))
+                raise ValueError(f'Uniform buffer is too small {buffer.size} is less than {size} for "{name}" with binding {binding}')
```

Comment:
for line length=120 let's try

msg = f'Uniform buffer is too small {buffer.size} is less than {size} for "{name}" with binding {binding}'
raise ValueError(msg)

if still too long i raise the line length limit for this file

Reply:
Ah, I didn't notice there was a line length check!